lib/email/base_email_unsubscriber.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+# This class and its children are instantiated and used by the EmailController.
+module Email
```

Comment:
I wonder if this should be `EmailControllerHelper` or something equivalent instead. `Email` feels too generic for something that is very specific to a controller.